frontend/packages/knative-plugin/src/components/overview/__tests__/KnativeOverview.spec.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import * as _ from 'lodash';
+import { PodRing } from '@console/shared';
+import { ResourceSummary } from '@console/internal/components/utils';
+import { KnativeOverview } from '../KnativeOverview';
+import { mockOverviewItemData } from './overview-knative-test-data';
+
+describe('KnativeOverview', () => {
+  it('should not render PodRing if obj.kind is not RevisionModel.kind', () => {
+    const wrapper = shallow(<KnativeOverview item={mockOverviewItemData.item} />);
+    expect(wrapper.find(PodRing)).toHaveLength(1);
+  });
+  it('should render ResourceSummary', () => {
+    const wrapper = shallow(<KnativeOverview item={mockOverviewItemData.item} />);
+    expect(wrapper.find(ResourceSummary)).toHaveLength(1);
+  });
+  it('should not render PodRing if obj.kind is not RevisionModel.kind', () => {
+    const mockItemKindRoute = _.set(mockOverviewItemData.item, 'obj.kind', 'Route');
```

Comment:
I'd deep clone this and not modify it directly. This is mutating an object reference and that can lead to trouble in future tests / any test that runs after this and uses the data.

Reply:
fixed